### PR TITLE
CMake and unit test machinery for developer host testing.

### PR DESCRIPTION
**Pull request by mookerji**
/cc @axlan @jacobmcnamee

### CODE REVIEW

**Comment by mookerji** [comment]
Pulls out cmake stuff from #154 and #159 .

File: board/piksiv3/rootfs-overlay/etc/monitrc [hunk 1]
```diff
@@ -46,4 +46,3 @@ check process syslogd with pidfile /var/run/syslogd.pid
 check process klogd with pidfile /var/run/klogd.pid
     start program = "/etc/init.d/S01logging start"
     stop program = "/etc/init.d/S01logging stop"
-
```
Comment (axlan) [inline board/piksiv3/rootfs-overlay/etc/monitrc]: Commit noise

Comment (mookerji) [inline board/piksiv3/rootfs-overlay/etc/monitrc]: Fixed

**Comment by mookerji** [comment]
TODO: 
- [ ] Get travis working
- [ ] Unfuck czmq path installation
- [ ] Cleanup 
- [ ] Review

File: run_tests.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+#
+# Script for running developer tests.
+
+set -e
+
+mkdir -p package/build && cd package/build
+cmake ../ && make
```
Comment (axlan) [inline run_tests.sh]: The default make includes 'make tests' ?

**Comment by axlan** [comment]
lgtm

File: run_tests.sh (hunk 2 above)
Comment (mookerji) [inline run_tests.sh]: I think once there are tests implemented, we should add tests as a default target to the cmake.

**Comment by mookerji** [comment]
Closed in favor or https://github.com/swift-nav/piksi_buildroot/pull/163 (merging from branch, not from personal branch).

**Pull request closed without merging**